### PR TITLE
[MOD-86][Fix] Fix scroll reset on route transition

### PR DESCRIPTION
**Pull request by sheriefvt**

### CODE REVIEW

File: app/router.js [hunk 1]
```diff
@@ -3,7 +3,19 @@ import config from './config/environment';
 
 const Router = Ember.Router.extend({
     location: config.locationType,
-    rootURL: config.rootURL
+    rootURL: config.rootURL,
+    willTransition(oldInfos, newInfos) {
+        const disableList = ['settings', 'moderation'];
```
Comment (aaxelb) [inline app/router.js]: This should be an all-caps constant at the top level of the file, with a more descriptive name (maybe `SCROLL_RESET_DISABLED_ROUTES`? or something more concise)

File: app/router.js [hunk 2]
```diff
@@ -3,7 +3,19 @@ import config from './config/environment';
...
+        const prevRoute = oldInfos.length ? oldInfos[oldInfos.length - 1].name.split('.') : null;
```
Comment (aaxelb) [inline app/router.js]: It'd be nicer to use `Ember.Router._routePath(infos)` (which I discovered looking at the [parent `willTransition`](https://github.com/emberjs/ember.js/blob/v2.14.1/packages/ember-routing/lib/system/router.js#L331) -- this feels like a case where using a private method is better than doing it manually), and have the list be `['preprints.provider.moderation', 'preprint.provider.settings']`

File: app/router.js [hunk 3]
```diff
@@ -3,7 +3,19 @@ import config from './config/environment';
...
+        const prevName = prevRoute != null ? prevRoute[prevRoute.length - 1] : null;
+        const currentRoute = newInfos.length ? newInfos[newInfos.length - 1].name.split('.') : null;
+        const currentName = currentRoute != null ? currentRoute[currentRoute.length - 1] : null;
+        const disableResetScroll = disableList.includes(prevName) && disableList.includes(currentName);
+        if (!disableResetScroll) {
+            this._super(...arguments);
```
Comment (aaxelb) [inline app/router.js]: I think it should always call `_super`, even if it doesn't scroll to the top. Unless there's something I'm missing?

**Review by aaxelb — CHANGES_REQUESTED** [review]
Looks good! Just a few cleanup suggestions.

**Comment by sheriefvt** [comment]
@aaxelb thanks for ur feedback, done with changes

File: app/router.js [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+const SCROLL_RESET_DISABLED_ROUTES = ['preprints.provider.moderation', 'preprints.provider.settings'];
+
 const Router = Ember.Router.extend({
     location: config.locationType,
-    rootURL: config.rootURL
+    rootURL: config.rootURL,
+    willTransition(oldInfos, newInfos) {
+        this._super(...arguments);
+        const prevPath = Ember.Router._routePath(oldInfos);  //Previous route
+        const newPath = Ember.Router._routePath(newInfos);   //New route
```
Comment (aaxelb) [inline app/router.js]: You can remove these two comments, I think it's pretty clear what these are. If you want, could rename the variables to `oldRoute` and `newRoute` or something like that.

**Review by aaxelb — DISMISSED** [review]
Looks good! 👍

**Comment by sheriefvt** [comment]
@aaxelb done

**Review by aaxelb — APPROVED** [review]

**Pull request merged**